src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+            }
+        }
+
+        void InsertManifestDependencies(
+            SQLite::Connection& connection,
+            SQLite::rowid_t manifestRowId,
+            std::set<Manifest::Dependency>& dependencies)
+        {
+            using namespace SQLite::Builder;
+            using namespace Schema::V1_0;
+
+            std::vector<Manifest::Dependency> missingPackageNodes;
+            std::map<Utility::Version, SQLite::rowid_t> versionsMap;
+            std::map<Manifest::Manifest::string_t, SQLite::rowid_t> idsMap;
+            for (const auto& dep : dependencies)
+            {
+                auto depMinVersion = dep.MinVersion.value();
+                auto packageId = dep.Id;
+
+                const auto packageRowId = IdTable::SelectIdByValue(connection, packageId);
+                if (!packageRowId.has_value())
+                {
+                    missingPackageNodes.emplace_back(dep);
+                    continue;
+                }
+
+                auto manifestVersions = ManifestTable::GetAllValuesById<IdTable, VersionTable>(connection, packageRowId.value());
+
+                auto result = std::find_if(
+                    manifestVersions.begin(),
+                    manifestVersions.end(),
+                    [&](auto& manifestVersion)
+                    {
+                        return (std::get<0>(manifestVersion) == depMinVersion.ToString());
+                    }
+                );
+
+                if (result == manifestVersions.end())
+                {
+                    VersionTable::EnsureExists(connection, depMinVersion.ToString());
+                }
+
+                idsMap[packageId] = packageRowId.value();
+                versionsMap[depMinVersion] = VersionTable::SelectIdByValue(connection, depMinVersion.ToString()).value();
+            }
+
+            ThrowOnMissingPackageNodes(missingPackageNodes);
+
+            StatementBuilder insertBuilder;
+            insertBuilder.InsertInto(s_DependenciesTable_Table_Name)
+                .Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name })
+                .Values(Unbound, Unbound, Unbound);
+            SQLite::Statement insert = insertBuilder.Prepare(connection);
+
+            for (const auto& dep : dependencies)
+            {
+                insert.Reset();
+                insert.Bind(1, manifestRowId);
+                insert.Bind(2, versionsMap[dep.MinVersion.value().ToString()]);
+                insert.Bind(3, idsMap[dep.Id]);
+
+                insert.Execute(true);
+            }
+        }
+
+        std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> GetPackageRowIdAndVersion(SQLite::Connection& connection, const Manifest::Dependency& dependency)
+        {
+            std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> row;
```

Comment:
>std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> row; [](http://example.com/codeflow?start=12&length=73)

Hint on `optional`, you can just return the value type when you have something and `{}` when you want it to be empty.